--- yersinia/src/admin.c	2024-09-30 19:43:14.906209490 -0400
+++ yersinia.new/src/admin.c	2024-09-30 19:21:22.252194755 -0400
@@ -330,13 +330,13 @@
    if (pthread_mutex_lock(&terms->mutex) != 0)
       thread_error("th_network_peer pthread_mutex_lock",errno);
 
-   fail = term_add_node(&term_node, TERM_VTY, sock, pthread_self());
+   fail = term_add_node(&term_node, TERM_VTY, (int)sock, pthread_self());
 
    if (fail == -1)
    {
       if (pthread_mutex_unlock(&terms->mutex) != 0)
          thread_error("th_network_peer pthread_mutex_unlock",errno);
-      admin_th_network_peer_exit(term_node, sock);
+      admin_th_network_peer_exit(term_node, (int)sock);
    }
     
    if (term_node == NULL)
@@ -346,7 +346,7 @@
       
       if (pthread_mutex_unlock(&terms->mutex) != 0)
          thread_error("th_network_peer pthread_mutex_unlock",errno);
-      admin_th_network_peer_exit(term_node, sock);
+      admin_th_network_peer_exit(term_node, (int)sock);
    }
 
    pthread_mutex_lock(&term_node->thread.finished);
@@ -377,14 +377,14 @@
       thread_error("getpeername",errno);
       if (pthread_mutex_unlock(&terms->mutex) != 0)
          thread_error("th_vty_peer pthread_mutex_unlock",errno);
-      admin_th_network_peer_exit(term_node, sock);
+      admin_th_network_peer_exit(term_node, (int)sock);
    }
 
     if (init_attribs(term_node) < 0)
     {
        if (pthread_mutex_unlock(&terms->mutex) != 0)
            thread_error("th_vty_peer pthread_mutex_unlock",errno);
-       admin_th_network_peer_exit(term_node, sock);
+       admin_th_network_peer_exit(term_node, (int)sock);
     }                               
    
    term_node->from_port = ntohs(name.sin_port);
